oauth2/src/main/java/com/okta/spring/boot/oauth/OktaOAuth2AutoConfig.java

```diff
@@ -70,6 +82,52 @@ OidcClientInitiatedLogoutSuccessHandler oidcLogoutSuccessHandler(OktaOAuth2Prope
         return new OktaOidcUserService(oAuth2UserService(authoritiesProviders), authoritiesProviders);
     }
 
+    @Bean
+    @ConditionalOnMissingBean
+    RestTemplate restTemplate(OktaOAuth2Properties oktaOAuth2Properties) {
+
+        Proxy proxy;
+
+        OktaOAuth2Properties.Proxy proxyProperties = oktaOAuth2Properties.getProxy();
+        if (proxyProperties != null && Strings.hasText(proxyProperties.getHost()) && proxyProperties.getPort() != 0) {
+            proxy = new Proxy(Proxy.Type.HTTP, new InetSocketAddress(proxyProperties.getHost(), proxyProperties.getPort()));
+
+            if (Strings.hasText(proxyProperties.getUsername()) &&
+                Strings.hasText(proxyProperties.getPassword())) {
+
+                Authenticator.setDefault(new ProxyPasswordAuthentication(proxyProperties.getUsername(),
+                    proxyProperties.getPassword().toCharArray()));
```

Comment:
Don't set the default authenticator, this needs to be scoped to this RestTemplate instance

Reply:
👍